openfe/protocols/openmm_md/plain_md_methods.py

```diff
@@ -490,58 +523,59 @@ def run(self, *, dry=False, verbose=True,
         solvent_comp, protein_comp, small_mols = system_validation.get_components(stateA)
 
         # 1. Create stateA system
-        # a. get a system generator
+        # Create a dictionary of OFFMol for each SMC for bookeeping
+        smc_components: dict[SmallMoleculeComponent, OFFMolecule]
+
+        smc_components = {i: i.to_openff() for i in small_mols}
+
+        # a. assign partial charges to smcs
+        self._assign_partial_charges(charge_settings, smc_components)
+
+        # b. get a system generator
         if output_settings.forcefield_cache is not None:
             ffcache = shared_basepath / output_settings.forcefield_cache
         else:
             ffcache = None
 
-        system_generator = system_creation.get_system_generator(
-            forcefield_settings=forcefield_settings,
-            integrator_settings=integrator_settings,
-            thermo_settings=thermo_settings,
-            cache=ffcache,
-            has_solvent=solvent_comp is not None,
-        )
-
-        smc_components: dict[SmallMoleculeComponent, OFFMolecule]
+        # Note: we block out the oechem backend for all systemgenerator
+        # linked operations to avoid any smiles operations that can
+        # go wrong when doing rdkit->OEchem roundtripping
+        with without_oechem_backend():
```

Comment:
I'm trying to be a bit conservative here by moving things around and avoiding using the context manager multiple times in repeat.

Long term this all needs a tidy up into a set of class methods that get called in a row (a bit like afe's base.py), so I'm not too fussed about pretty-ness. If someone cares, please do hack away at this!